src/utils/records.ts

```diff
@@ -263,6 +265,11 @@ export class Records {
       filterCopy.datePublished = datePublishedFilter;
     }
 
+    //if sorting by published, results must filter for published
+    if (dateSort === DateSort.PublishedAscending || dateSort === DateSort.PublishedDescending) {
+      filterCopy.published = true;
+    }
```

Comment:
This check feels strange to be part of "convertFilter", consider having it else where.

Reply:
Discussed offline, @LiranCohen tried to pull it out elsewhere but it didn't make much sense on it's own. Since it's converting the types the same thing would need to happen in several places after the conversion (or adding to the input).